dpp/src/tests/identity/state_transition/identity_update_transition/identity_update_transition_spec.rs

```diff
@@ -0,0 +1,221 @@
+    );
+}
+
+#[test]
+fn set_public_keys_to_add() {
+    let TestData { mut transition, .. } = setup_test();
+    let id_public_key = IdentityPublicKey {
+						id : 0,
+            key_type: KeyType::BLS12_381,
+            purpose: Purpose::AUTHENTICATION,
+            security_level : SecurityLevel::CRITICAL,
+            read_only: true,
+            data: hex::decode("01fac99ca2c8f39c286717c213e190aba4b7af76db320ec43f479b7d9a2012313a0ae59ca576edf801444bc694686694").unwrap(),
+            disabled_at : None,
+        };
+    transition.set_public_keys_to_add(vec![id_public_key.clone()]);
+
+    assert_eq!(vec![id_public_key], transition.get_public_keys_to_add());
+}
+
+#[test]
+fn get_disable_public_keys() {
+    let TestData { transition, .. } = setup_test();
+    assert_eq!(
+        transition.disable_public_keys,
+        transition.get_public_key_ids_to_disable()
+    );
+}
+
+#[test]
+fn set_disable_public_keys() {
+    let TestData { mut transition, .. } = setup_test();
+    let id_to_disable = vec![1, 2];
+    transition.set_public_key_ids_to_disable(id_to_disable.clone());
+
+    assert_eq!(&id_to_disable, transition.get_public_key_ids_to_disable());
+}
+
+#[test]
+fn get_public_key_disabled_at() {
+    let TestData { transition, .. } = setup_test();
+    assert_eq!(
+        transition.public_keys_disabled_at,
+        transition.get_public_keys_disabled_at()
+    );
+}
+
+#[test]
+fn set_public_key_disabled_at() {
+    let TestData { mut transition, .. } = setup_test();
+    let now = Utc::now().timestamp_millis() as u64;
+    transition.set_public_keys_disabled_at(Some(now));
+
+    assert_eq!(Some(now), transition.get_public_keys_disabled_at());
+}
+
+#[test]
+fn to_object() {
+    let TestData { transition, .. } = setup_test();
+    let result = transition
+        .to_object(false)
+        .expect("conversion to object shouldn't fail");
+
+    let expected_raw_state_transition = json!({
+        "protocolVersion" : 1,
+        "type" : 5,
+        "signature" : [],
+        "signaturePublicKeyId": 0,
+        "identityId" : transition.identity_id.to_buffer(),
+        "revision": 0,
+        "disablePublicKeys" : [0],
+        "publicKeysDisabledAt" : 1234567,
+        "addPublicKeys" : [
+            {
+
+                "id" : 3,
+                "purpose" : 0,
+                "type": 0,
+                "securityLevel" : 0,
+                "data" :base64::decode("AkVuTKyF3YgKLAQlLEtaUL2HTditwGILfWUVqjzYnIgH").unwrap(),
+                "readOnly" : false
+            }
+        ]
+    });
+
+    assert_eq!(expected_raw_state_transition, result);
+}
+
+#[test]
+fn to_object_with_signature_skipped() {
+    let TestData { transition, .. } = setup_test();
+    let result = transition
+        .to_object(true)
+        .expect("conversion to object shouldn't fail");
+
+    let expected_raw_state_transition = json!({
+        "protocolVersion" : 1,
+        "type" : 5,
+        "signaturePublicKeyId": 0,
```

Comment:
Should the id still be there if we've removed the signature? I've actually never thought about this before - in JS DPP it seems we do the same thing. Probably something to discuss at the architecture meeting, @shumkov @jawid-h @shuplenkov

Reply:
I guess the logic was that when you call the `.sign` method in JS this field is unset. Not sure

Reply:
(This comment shouldn't prevent the PR from being merged, as it is in line with the original implementation)